contracts/core/Data/SportsAMMV2Data.sol

```diff
@@ -163,6 +163,39 @@ contract SportsAMMV2Data is Initializable, ProxyOwned, ProxyPausable {
         }
     }
 
+    /**
+     * @notice Retrieves free bets data for a specific user.
+     * @dev Fetches free bets data for a given user and collateral addresses.
+     * @param user The address of the user.
+     * @param collateralAddresses An array of collateral addresses to retrieve free bets data for.
+     * @return freeBetsAmountPerCollateral An array of free bets amounts for each collateral address.
+     * @return freeBetsExpiryPerCollateral An array of free bets expiries for each collateral address.
+     */
+    function getFreeBetsDataPerUser(
+        address user,
+        address[] memory collateralAddresses
+    ) external view returns (uint[] memory freeBetsAmountPerCollateral, uint[] memory freeBetsExpiryPerCollateral) {
+        freeBetsAmountPerCollateral = new uint[](collateralAddresses.length);
+        freeBetsExpiryPerCollateral = new uint[](collateralAddresses.length);
+        IFreeBetsHolder freeBetsHolder = sportsAMM.freeBetsHolder();
+        for (uint i = 0; i < collateralAddresses.length; i++) {
+            freeBetsAmountPerCollateral[i] = freeBetsHolder.balancePerUserAndCollateral(user, collateralAddresses[i]);
+            uint userFreeBetExpiration = freeBetsHolder.freeBetExpiration(user, collateralAddresses[i]);
+            if (userFreeBetExpiration == 0) {
+                userFreeBetExpiration = freeBetsHolder.freeBetExpirationUpgrade() +
```

Comment:
freeBetsHolder.freeBetExpirationUpgrade()  is always the same, no need to call it in every iteration